File: mac/Tests/BG3AssistantTests/ActLedgerTests.swift
import XCTest
@testable import BG3HonorAssistant

final class ActLedgerTests: XCTestCase {
    func testOnlyPastActLedgersAreLocked() {
        var run = HonorRun()
        run.selectedAct = 2

        XCTAssertTrue(run.actLedgerIsLocked(1))
        XCTAssertFalse(run.actLedgerIsLocked(2))
        XCTAssertFalse(run.actLedgerIsLocked(3))
    }

    func testFinalActRecordLocksActThreeLedger() {
        var run = HonorRun()
        run.selectedAct = 3
        run.finalActRecord = ActTransitionRecord(
            fromAct: 3,
            toAct: 3,
            gearReview: ["item": .obtained],
            unresolvedRouteCount: 0,
            advancedAt: Date(timeIntervalSince1970: 1)
        )

        XCTAssertTrue(run.actLedgerIsLocked(3))
        XCTAssertEqual(run.lockedActGearReviewStatus(for: "item", in: 3), .obtained)
    }
}

File: windows/tests/BG3HonorAssistant.Core.Tests/Models/ActLedgerTests.cs
using BG3HonorAssistant.Core.Models;

namespace BG3HonorAssistant.Core.Tests.Models;

public sealed class ActLedgerTests
{
    [Fact]
    public void OnlyPastActLedgersAreLocked()
    {
        var run = new HonorRun { SelectedAct = 2 };

        Assert.True(run.ActLedgerIsLocked(1));
        Assert.False(run.ActLedgerIsLocked(2));
        Assert.False(run.ActLedgerIsLocked(3));
    }

    [Fact]
    public void FinalActRecordLocksActThreeLedger()
    {
        var run = new HonorRun
        {
            SelectedAct = 3,
            FinalActRecord = new ActTransitionRecord(
                3,
                3,
                new Dictionary<string, ActGearReviewStatus>
                {
                    ["item"] = ActGearReviewStatus.Obtained,
                },
                0,
                DateTimeOffset.FromUnixTimeSeconds(1)),
        };

        Assert.True(run.ActLedgerIsLocked(3));
        Assert.Equal(
            ActGearReviewStatus.Obtained,
            run.LockedActGearReviewStatus("item", 3));
    }
}
